install_clang.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -eux &&
+
+# Building clang requires cmake (and we have ninja there too at that point)
+PATH="$HOME/misc/root/cmake/bin:$PATH"
+
+readonly url='https://github.com/llvm/llvm-project/archive/c0da957bb05b7044bec83f8159c57b1d9ab9eb59.tar.gz'
+
+readonly workdir="$HOME/misc/code/clang"
+readonly srcdir="$workdir/clang"
+readonly builddir="$workdir/build"
+readonly jobs=$(($(getconf _NPROCESSORS_ONLN) * 1))
+
+curl -fLs "$url" | tar xz --strip-components=1 --one-top-level="$srcdir"
+mkdir -p "$builddir"
+cd "$builddir"
+cmake -G Ninja \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DCMAKE_INSTALL_PREFIX="$HOME/misc/root/clang" \
+	-DLLVM_ENABLE_PROJECTS='clang' \
```

Comment:
I think we also want a shared build. That should certainly reduce the installation tree size. I've found that `-DLLVM_PARALLEL_LINK_JOBS:STRING=1` was required to not blow memory limits when linking. Adding `-DLLVM_INCLUDE_TESTS:BOOL=OFF` would also likely reduce build times.

Reply:
What's the option to enable shared build only? I've heard something about libllvm.so.
There may be other options that may reduce build time and image size, like selecting only the host machine's architecture (I think by default it's building a bunch of cross-compiling support as I can see it building stuff related to many archs in the log). Other options I don't know may limit which languages are supported.

Reply:
`-DBUILD_SHARED_LIBS:BOOL=ON`

I think removing cross compilation makes sense. Removing TableGen compilation would certainly help compilation time. Likely link time/memory too.

Reply:
OK. I got the install size of clang reduced by 10 times! (from 1.5GB down to 155MB). I played with the LVM_INCLUDE_TESTS and LVM_BUILD_TESTS options but it triggers a bug where a link commands wants the gtest lib, and it's not there. Anyway, the doc says LVM_BUILD_TESTS is OFF by default (the LVM_INCLUDE_TESTS seems to only control whether or not the generated ninja file includes rules for the test target or not).

Reply:
Sweet! I'm looking at reducing the GCC install size now. Probably don't need to compile non-C and C++ compilers…